ob_intern/Controller/Mask Video/StampLogoVideoViewController.swift

```diff
@@ -0,0 +1,350 @@
+        buttonConfirm.isHidden = true
+        progressBar.setProgress(0.0, animated: false)
+
+    }
+
+    @objc func updateExportDisplay(_ timer: Timer) {
+        guard let progressBar = timer.userInfo as? UIProgressView, let exporter = exporter else { return }
+
+        let progress = exporter.progress
+        print("progress \(progress)")
+        
+        // Update the progress bar on the main queue
+        DispatchQueue.main.async {
+            progressBar.setProgress(progress, animated: true)
+            print("progress in update \(progressBar.progress)")
+        }
+    }
+    
+    func setupVideoPicker() {
+        let picker = UIImagePickerController()
+        picker.delegate = self
+        picker.sourceType = .photoLibrary
+        picker.delegate = self
+        picker.mediaTypes = UIImagePickerController.availableMediaTypes(for: .photoLibrary) ?? []
+        picker.mediaTypes = ["public.movie"]
+        picker.videoQuality = .typeHigh
+        //picker.videoExportPreset = AVAssetExportPresetHEVC1920x1080
+        picker.allowsEditing = false
+        present(picker, animated: true, completion: nil)
+        
+    }
+    
+    func setupDragAndDrop() {
+        addTextView = TextEditDragDropViewController.init(nibName: "TextEditDragDropViewController", bundle: nil)
+//        let width = UIScreen.main.bounds.width
+//        let height = width * (16/9)
+//        addTextView?.view.frame = CGRect(x: 0, y: 0, width: width, height: height)
+        addTextView?.delegate = self
+        if let addTextView = self.addTextView {
+            view.addSubview(addTextView.view)
+        }
+    }
+    
+    func createVideoPlayer(url: URL) {
+        player = AVPlayer(url: url)
+        playerLayer = AVPlayerLayer(player: player)
+//        playerLayer.videoGravity = .resizeAspectFill
+        playerLayer.frame = viewVideo.bounds
+        viewVideo.layer.addSublayer(playerLayer)
+        
+    }
+    
+    func createVideoController(playerItem: AVPlayerItem) {
+        player = AVPlayer(playerItem: playerItem)
+        print("playerItem \(playerItem)")
+        playerViewController.player = player
+        self.present(playerViewController, animated: true) {
+            self.playerViewController.player!.play()
+        }
+    }
+    
+    func addComposition(asset: AVAsset, image: UIImage, url: URL) {
+        let composition = AVMutableVideoComposition(asset: asset) { request in
+            self.customText = CIImage(image: image)
+            
+            let positionedText = self.customText?.transformed(by: .identity, highQualityDownsample: true)
+            
+            //background
+            let backgroundColorFilter = CIFilter(name: "CIConstantColorGenerator")!
+            let backgroundColor = CIColor(cgColor: UIColor.black.cgColor)
+            backgroundColorFilter.setValue(backgroundColor, forKey: kCIInputColorKey)
+            let screenSize = CGRect(x: 0, y: 0, width: 1080, height: 1920)
```

Comment:
```suggestion
            let screenSize = CGRect(x: 0, y: 0, width: 1080, height: 1920)
```
ฝากเอา 1920 ไปทำเป็น VARIABLE ไว้ครับ